instance/tests/integration/factories/instance.py

```diff
@@ -23,12 +23,25 @@
 # Imports #####################################################################
 
 import uuid
+from random import randint
 
 import factory
 from factory.django import DjangoModelFactory
+from django.conf import settings
 
 from instance.models.mixins.domain_names import generate_internal_lms_domain
 from instance.models.openedx_instance import OpenEdXInstance
+from instance.models.utils import ConsulAgent
+
+
+def _get_unused_id():
+    """
+    Find an unused ID for Consul
+    """
+    while True:
+        i = randint(2 ** 30, 2 ** 31)
+        if not ConsulAgent(prefix=settings.CONSUL_PREFIX.format(ocim=settings.OCIM_ID, instance=i)).get(''):
```

Comment:
If  `Consul` isn't available, this errors out with 
```
HTTPConnectionPool(host='127.0.0.1', port=8500): Max retries exceeded with url: /v1/kv/ocim/instances/1183429413 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f135e757208>: Failed to establish a new connection: [Errno 111] Connection refused',))
```

Can you catch this exception and return the first generated number when this happens?

Reply:
Sure! I didn't expect it to be used without Consul but yeah, better handle that gracefully.